[Low Wei Bin] iP

OrangeCatLoves opened the pull request:
# TheOrangeRatchetCat  🐱 🐈 
> "Meow"

MEOW ~ I CAN DO ALL THESE FOR YOU 💯 
- Store your tasks
- Mark/Unmark tasks base on completion
- Display all tasks
- Display tasks on relevant dates
- Find matching tasks

All you need to do is:
   1. Download it from [here](https://github.com/OrangeCatLoves/ip/releases/tag/A-Gradle)
   2. Run the program
   3. Input your tasks
   4. Let TheOrangeRatchetCat manage it for you!
  
IT IS ALSO FREE! 👍 

Features:

- [x] Task Management
- [ ] Task Reminder
- [ ] Task Calender

---
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class TheOrangeRatchetCat {
    public static void main(String[] args) {
        TheOrangeCat.run();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

OrangeCatLoves committed:
Level-0

OrangeCatLoves committed:
Level-0 v2

OrangeCatLoves committed:
Level-1 v1

OrangeCatLoves committed:
Level-3 v1

OrangeCatLoves committed:
Level-3 v1

OrangeCatLoves committed:
Level-4 v1

OrangeCatLoves committed:
A-TextUiTesting v1

OrangeCatLoves committed:
Level-5 v1

OrangeCatLoves committed:
Level-6 v1

OrangeCatLoves committed:
A-Enums v1

OrangeCatLoves committed:
Level-5 v2

OrangeCatLoves committed:
Level-6 v2

OrangeCatLoves committed:
Level-7 v1

OrangeCatLoves committed:
Level-8 v1

OrangeCatLoves committed:
A-MoreOOP v1

OrangeCatLoves committed:
Merge remote-tracking branch 'origin/add-gradle-support'

OrangeCatLoves committed:
A-Packages v1

OrangeCatLoves committed:
A-Gradle v1

OrangeCatLoves committed:
A-JUnit v1

OrangeCatLoves committed:
A-Jar v1

OrangeCatLoves committed:
A-JavaDoc v1

OrangeCatLoves committed:
A-CodingStandard v1

OrangeCatLoves committed:
Level-9 v1

OrangeCatLoves committed:
Level-9 v2

OrangeCatLoves committed:
Created Command class + Fixed some code standards

OrangeCatLoves committed:
CheckStyle-v1

OrangeCatLoves committed:
Level-10 v1

Jasperabez commented on src/main/java/TheOrangeRatchetCat.java:
```diff
@@ -0,0 +1,63 @@
+import parser.Command;
+import storage.Storage;
+import parser.Parser;
+import java.util.List;
+import java.util.ArrayList;
```
Could import order be more consistent throughout different files? See `TheOrangeCat.java` lines 7-8.

Jasperabez commented on src/main/java/parser/OnCommand.java:
```diff
@@ -0,0 +1,22 @@
+    public String execute(String input,List<Task> items, Scanner scanner) {
+        try {
+            LocalDate localDate = LocalDate.parse(input.substring(2).trim());
+            return TaskList.activitiesOnThisDate(localDate, items, scanner);
```
Would `activitiesOnThisDate` be better named as a verb instead?

Jasperabez commented on src/main/java/tasklist/TaskList.java:
```diff
@@ -0,0 +1,324 @@
+     * @param scanner the scanner object to read next line of input
+     * @return returns the next input command by user
+     */
+    public static String checkList(List<Task> items, Scanner scanner) {
```
Is there a reason why scanner is part of the arguments but not used in the function body? And is there a better name to be given for checkList based on your javadocs descriptor?

Jasperabez commented on src/main/java/tasklist/TaskList.java:
```diff
@@ -0,0 +1,324 @@
+     * @param scanner the scanner object to read next line of input
+     * @return returns the next input command by user
+     */
+    public static String markingTask(String input, List<Task> items, Scanner scanner) {
```
Would using imperative naming e.g `markTask` be more preferable in this scenario? Same for the other functions in this class.

Jasperabez commented on src/main/java/tasklist/TaskList.java:
```diff
@@ -0,0 +1,324 @@
+    public static String markingTask(String input, List<Task> items, Scanner scanner) {
+        try {
+            int taskIndex = Integer.parseInt(input) - 1;
+            Task markingTask = items.get(taskIndex);
```
Would variables be better named as nouns instead of verbs?

Jasperabez commented on src/main/java/tasklist/TaskList.java:
```diff
@@ -0,0 +1,324 @@
+        try {
+            int taskIndex = Integer.parseInt(input) - 1;
+            Task markingTask = items.get(taskIndex);
+            markingTask.setDone(false);
```
Any reason why this is named `markingTask` instead of something like `unmarkingTask`?

Jasperabez commented on src/main/java/tasklist/TaskList.java:
```diff
@@ -0,0 +1,324 @@
+     * @return returns the next input command by user
+     */
+    // Make sure you deal with incorrect inputs, couldn't load => on 2024-08
+    public static String activitiesOnThisDate(LocalDate date, List<Task> items, Scanner scanner) {
```
Would this be better named as a verb?

Jasperabez commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,46 @@
+        return "[" + this.getStatusIcon() + "] " + this.description;
+    }
+
+    public void setDone(boolean done) { this.isDone = done; }
```
Egyptian-style indentation would be preferred according to the coding standard. Would this have been better? `this.isDone = done` in new line.

Jasperabez commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,46 @@
+
+    public boolean getDoneStatus() { return this.isDone; }
+
+    public String getDes() { return this.description; }
```
Would keeping Description in its full name be better for readability? e.g `getDescription()`

Jasperabez reviewed (commented):
LGTM except some nitpicks on the import order consistency, and naming conventions

TerenceWongsExtravaganza commented on src/main/java/orangecat/TheOrangeCat.java:
I like the use of enums

TerenceWongsExtravaganza commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,72 @@
+    // To initialise all the commands inside the map
+    public static void initialiseMap() {}
+
+    public static void ratchetCatBot(List<Task> items) {
```
is the naming of this function considered to be in compliance with the standard?

TerenceWongsExtravaganza commented on src/main/java/tasklist/TaskList.java:
```diff
@@ -0,0 +1,324 @@
+     */
+
+    // Deal with incorrect inputs todo <space> does not terminate
+    public static String addingToDo(String input, List<Task> items, Scanner scanner) throws TheOrangeRatchetCatException {
```
is the variable name imperative?

TerenceWongsExtravaganza commented on src/main/java/ui/Ui.java:
```diff
@@ -0,0 +1,60 @@
+        System.out.println("____________________________________________________________");
+    }
+
+    public static void sayWelcome() {
```
nice ascii art

OrangeCatLoves committed:
Added a new method addMany that allow users to add multiple tasks at one go in a single command

OrangeCatLoves committed:
Refactored the code with new checkstyle after using checkstyle plugin

OrangeCatLoves committed:
Removed irrelevant comments so that the code is cleaner

OrangeCatLoves committed:
Added some changes to the GUI design plus added some code for colouring of GUI

TerenceWongsExtravaganza reviewed (commented):
Looks good!

OrangeCatLoves committed:
Add assert statements to check assumptions in the code

Added assert statements in key functions to validate assumptions made about input values and program state. This helps improve code reliability and maintainability by catching potential errors early in the execution. For instance, added assertions to check that input values are non-negative and that certain variables are initialized before use.

These changes enhance documentation by making assumptions explicit within the code, providing developers with clearer expectations about function inputs and outputs. This is particularly useful in areas where incorrect inputs could lead to silent failures or incorrect results.

No changes to the main functionality; this is purely an improvement in code safety and readability.

OrangeCatLoves committed:
Add assert statements to check assumptions in the code

Added assert statements in key functions to validate assumptions made about input values and program state. This helps improve code reliability and maintainability by catching potential errors early in the execution. For instance, added assertions to check that input values are non-negative and that certain variables are initialized before use.

These changes enhance documentation by making assumptions explicit within the code, providing developers with clearer expectations about function inputs and outputs. This is particularly useful in areas where incorrect inputs could lead to silent failures or incorrect results.

No changes to the main functionality; this is purely an improvement in code safety and readability.

OrangeCatLoves committed:
Refactor code for improved readability and maintainability

Refactored several functions and code blocks to enhance readability and make the codebase easier to understand. Changes include renaming variables for clarity, breaking down complex functions into smaller, more manageable ones, and reorganizing code to follow a more logical flow.

These modifications do not change the functionality of the application but aim to improve the overall maintainability of the code. By making the code easier to read, future modifications and debugging will be more straightforward, reducing the cognitive load on developers working on the project.

The refactoring also adheres to the established coding standards and best practices, ensuring consistent style and enhancing code quality across the project.

OrangeCatLoves committed:
Refactor code for improved readability and maintainability

Refactored several functions and code blocks to enhance readability and make the codebase easier to understand. Changes include renaming variables for clarity, breaking down complex functions into smaller, more manageable ones, and reorganizing code to follow a more logical flow.

These modifications do not change the functionality of the application but aim to improve the overall maintainability of the code. By making the code easier to read, future modifications and debugging will be more straightforward, reducing the cognitive load on developers working on the project.

The refactoring also adheres to the established coding standards and best practices, ensuring consistent style and enhancing code quality across the project.

OrangeCatLoves committed:
Merge pull request #1 from OrangeCatLoves/branch-A-CodeQuality

PR from branch-A-CodeQuality to master

OrangeCatLoves committed:
Refactor code for readability and add assert statements for validation

Performed a comprehensive refactor of the codebase to enhance readability and maintainability. Key changes include renaming variables for clarity, breaking down complex functions into smaller, more focused units, and reorganizing code to follow a more logical structure. These improvements aim to make the code easier to understand and maintain.

In addition to the refactoring, added assert statements to verify critical assumptions and ensure data integrity throughout the application. These assertions check for valid input values, non-null variables, and other key conditions that must be met for the code to function correctly. The inclusion of these assertions helps catch potential issues early, providing better documentation and error handling.

Overall, these changes contribute to improved code quality and robustness, making future development and debugging more efficient.

OrangeCatLoves committed:
ci: set up GitHub Actions for continuous integration

Added a GitHub Actions workflow to automate CI processes. This setup includes tasks for building, testing, and validating the codebase to ensure consistent quality and catch issues early in the development cycle.

OrangeCatLoves committed:
feat: Implement prioritization of items with priority levels

- Added the ability to attach priority levels to items, allowing users to mark items with different priorities (e.g., high priority, priority level 1).
- Introduced an enumeration (enum) for defining priority levels, such as LOW, MEDIUM, HIGH, and CRITICAL.
- Updated task classes (Deadline, Todo, Event) to include a priority attribute.
- Modified the input handling and processing logic to accept and set priority levels for tasks.
- Enhanced UI to display priority levels for each task.

This feature allows users to better organize and prioritize their tasks based on importance, improving overall task management.

OrangeCatLoves committed:
Refactor Code for Improved Readability and Maintainability

- Extracted repeated logic into separate methods to improve clarity and reduce duplication.
- Renamed variables for better understanding of their roles and purposes.
- Simplified conditional statements and loops to enhance readability and efficiency.
- Added or updated comments to better explain complex code sections and logic.

Details:
- Consolidated redundant logic in [specific class or method name].
- Improved method names to reflect their actions more accurately.
- Reduced code complexity by refactoring large methods into smaller, more manageable ones.

Impact:
- The refactor improves the maintainability of the codebase and makes future modifications easier.
- No functional changes; the behavior of the application remains consistent.

OrangeCatLoves committed:
Replaced traditional loops and manual collection operations with Java Streams.
Simplified data processing and transformation tasks, enhancing code readability.
Improved performance by leveraging parallel streams where appropriate.
Updated related unit tests to reflect changes and ensure functionality.
This refactor enhances maintainability and leverages modern Java features for more efficient and expressive code.

OrangeCatLoves committed:
Simplify and optimize data processing functions
The current codebase contains complex and repetitive logic in data processing functions, making the code difficult to maintain and understand.

The complexity leads to a higher likelihood of bugs and makes future enhancements more challenging.

Refactor the data processing functions to remove redundant code, simplify the logic, and improve readability.

This approach reduces code duplication and increases maintainability, ensuring that the code is easier to work with for future development.

Unit tests have been updated to reflect the changes, and the refactoring has been done without altering the existing functionality.

OrangeCatLoves committed:
The codebase contains several comments that are redundant or do not add significant value, leading to visual clutter and reduced readability.

These unnecessary comments increase maintenance complexity and make the codebase harder to navigate, detracting from the overall cleanliness and simplicity of the code.

Remove redundant comments to enhance code readability and maintainability. This change streamlines the code, making it easier to understand at a glance without the distraction of excessive commentary.

This approach is chosen to ensure that only meaningful and essential comments remain, aligning with clean code principles and best practices.

OrangeCatLoves committed:
The codebase contains several comments that are redundant or do not add significant value, leading to visual clutter and reduced readability.

These unnecessary comments increase maintenance complexity and make the codebase harder to navigate, detracting from the overall cleanliness and simplicity of the code.

Remove redundant comments to enhance code readability and maintainability. This change streamlines the code, making it easier to understand at a glance without the distraction of excessive commentary.

This approach is chosen to ensure that only meaningful and essential comments remain, aligning with clean code principles and best practices.

OrangeCatLoves committed:
The codebase contains several comments that are redundant or do not add significant value, leading to visual clutter and reduced readability.

These unnecessary comments increase maintenance complexity and make the codebase harder to navigate, detracting from the overall cleanliness and simplicity of the code.

Remove redundant comments to enhance code readability and maintainability. This change streamlines the code, making it easier to understand at a glance without the distraction of excessive commentary.

This approach is chosen to ensure that only meaningful and essential comments remain, aligning with clean code principles and best practices.

OrangeCatLoves committed:
The codebase contains several comments that are redundant or do not add significant value, leading to visual clutter and reduced readability.

These unnecessary comments increase maintenance complexity and make the codebase harder to navigate, detracting from the overall cleanliness and simplicity of the code.

Remove redundant comments to enhance code readability and maintainability. This change streamlines the code, making it easier to understand at a glance without the distraction of excessive commentary.

This approach is chosen to ensure that only meaningful and essential comments remain, aligning with clean code principles and best practices.

OrangeCatLoves committed:
Refactor redundant code for clarity and maintenance

Currently, the codebase contains redundant blocks of code that perform similar operations across multiple classes, leading to increased complexity and potential maintenance issues. This redundancy makes it difficult to update or debug the code, as changes need to be replicated in multiple places. To improve clarity and maintainability, consolidate these redundant code blocks into shared methods within a utility class. This refactoring reduces duplication and centralizes common logic, making future updates more straightforward and less error-prone. This approach was chosen to streamline the code and ensure consistency across the application, thereby lowering the risk of introducing bugs during future enhancements.

OrangeCatLoves committed:
Refactor redundant code for clarity and maintenance

Currently, the codebase contains redundant blocks of code that perform similar operations across multiple classes, leading to increased complexity and potential maintenance issues. This redundancy makes it difficult to update or debug the code, as changes need to be replicated in multiple places. To improve clarity and maintainability, consolidate these redundant code blocks into shared methods within a utility class. This refactoring reduces duplication and centralizes common logic, making future updates more straightforward and less error-prone. This approach was chosen to streamline the code and ensure consistency across the application, thereby lowering the risk of introducing bugs during future enhancements.

OrangeCatLoves committed:
Refactor redundant code for clarity and maintenance

Currently, the codebase contains redundant blocks of code that perform similar operations across multiple classes, leading to increased complexity and potential maintenance issues. This redundancy makes it difficult to update or debug the code, as changes need to be replicated in multiple places. To improve clarity and maintainability, consolidate these redundant code blocks into shared methods within a utility class. This refactoring reduces duplication and centralizes common logic, making future updates more straightforward and less error-prone. This approach was chosen to streamline the code and ensure consistency across the application, thereby lowering the risk of introducing bugs during future enhancements.

OrangeCatLoves committed:
Refactor redundant code for clarity and maintenance

Currently, the codebase contains redundant blocks of code that perform similar operations across multiple classes, leading to increased complexity and potential maintenance issues. This redundancy makes it difficult to update or debug the code, as changes need to be replicated in multiple places. To improve clarity and maintainability, consolidate these redundant code blocks into shared methods within a utility class. This refactoring reduces duplication and centralizes common logic, making future updates more straightforward and less error-prone. This approach was chosen to streamline the code and ensure consistency across the application, thereby lowering the risk of introducing bugs during future enhancements.

OrangeCatLoves committed:
Refactor redundant code for clarity and maintenance

Currently, the codebase contains redundant blocks of code that perform similar operations across multiple classes, leading to increased complexity and potential maintenance issues. This redundancy makes it difficult to update or debug the code, as changes need to be replicated in multiple places. To improve clarity and maintainability, consolidate these redundant code blocks into shared methods within a utility class. This refactoring reduces duplication and centralizes common logic, making future updates more straightforward and less error-prone. This approach was chosen to streamline the code and ensure consistency across the application, thereby lowering the risk of introducing bugs during future enhancements.

OrangeCatLoves committed:
Update GUI for improved visuals and aesthetics

The current GUI lacks visual appeal and modern design elements, which could detract from user experience.

Enhance the interface by updating color schemes, adding animations, and refining layout for a more polished and cohesive look. This was done to align with user feedback and ensure the UI is visually engaging while maintaining functionality.

These changes improve overall aesthetics and create a more user-friendly experience, contributing to the app's professional appearance. This approach was taken to balance both form and function without overcomplicating the interface.

OrangeCatLoves committed:
Add personality to chatbot for more engaging interactions

Current interactions with the chatbot are functional but lack personality, resulting in a less engaging user experience.

Introduce conversational tone, dynamic responses, and personalized language to give the chatbot a distinct and engaging personality. This was done to make interactions feel more human and relatable, improving user satisfaction and overall experience.

The approach balances personality with maintaining the chatbot's original functionality, ensuring that the chatbot remains efficient while being more interactive and approachable.

OrangeCatLoves committed:
Improve error handling for chatbot to enhance robustness

Current chatbot error handling is insufficient, leading to unhandled exceptions and unclear feedback for users during failures.

Implement more comprehensive error handling to catch exceptions and provide meaningful error messages. This change ensures that users receive clearer guidance when something goes wrong and prevents the chatbot from crashing unexpectedly.

This approach improves the chatbot’s reliability and user experience by ensuring that errors are handled gracefully, with proper logging to aid in future debugging and issue resolution.

OrangeCatLoves committed:
Add new Ui.png mockup to illustrate proposed GUI design

Currently, there is no visual reference for the updated GUI, which makes it harder to communicate the design vision.

Add Ui.png as a mockup to provide a clear visual representation of the intended GUI layout and aesthetic changes. This image serves as a reference for developers and designers to align on the final design implementation.

Including this mockup helps ensure that the design vision is followed and facilitates better collaboration and feedback during development.

OrangeCatLoves committed:
Update README.md

OrangeCatLoves committed:
Update README.md

OrangeCatLoves committed:
Update README.md

OrangeCatLoves committed:
Update README.md

OrangeCatLoves committed:
Update README.md

OrangeCatLoves committed:
Update README.md

OrangeCatLoves committed:
Update README.md

OrangeCatLoves committed:
Update README.md

OrangeCatLoves committed:
Update README.md

OrangeCatLoves committed:
Update README.md

OrangeCatLoves committed:
Update README.md

OrangeCatLoves committed:
Update README.md

OrangeCatLoves committed:
Update README.md

OrangeCatLoves committed:
Update README.md

OrangeCatLoves committed:
Update README.md

OrangeCatLoves committed:
Update README.md

OrangeCatLoves committed:
Update README.md

OrangeCatLoves committed:
Add user guide for chatbot usage

The chatbot currently lacks clear instructions for users, leading to potential confusion about its functionalities.

Create a comprehensive user guide detailing how to use the chatbot, including features, commands, and example interactions. This guide is added to help users understand and navigate the chatbot effectively, improving overall user experience and engagement.

Including this documentation ensures users can fully utilize the chatbot and reduces potential friction during interactions.

OrangeCatLoves committed:
Merge branch 'master' of https://github.com/OrangeCatLoves/ip

OrangeCatLoves committed:
Refactor code to improve readability and maintainability

The current code contains redundant logic and complex methods,
making it difficult to manage.

Refactor by splitting large functions, removing repeated code,
and enhancing variable names for clarity.

This improves code readability and reduces complexity for easier
future updates. No functional changes were made.

OrangeCatLoves committed:
Refactor code to improve readability and maintainability

The current code contains redundant logic and complex methods,
making it difficult to manage.

Refactor by splitting large functions, removing repeated code,
and enhancing variable names for clarity.

This improves code readability and reduces complexity for easier
future updates. No functional changes were made.

OrangeCatLoves committed:
Refactor code to improve readability and maintainability

The current code contains redundant logic and complex methods,
making it difficult to manage.

Refactor by splitting large functions, removing repeated code,
and enhancing variable names for clarity.

This improves code readability and reduces complexity for easier
future updates. No functional changes were made.

OrangeCatLoves committed:
Refactor code to improve readability and maintainability

The current code contains redundant logic and complex methods,
making it difficult to manage.

Refactor by splitting large functions, removing repeated code,
and enhancing variable names for clarity.

This improves code readability and reduces complexity for easier
future updates. No functional changes were made.

OrangeCatLoves committed:
Refactor code to improve readability and maintainability

The current code contains redundant logic and complex methods,
making it difficult to manage.

Refactor by splitting large functions, removing repeated code,
and enhancing variable names for clarity.

This improves code readability and reduces complexity for easier
future updates. No functional changes were made.

OrangeCatLoves committed:
Refactor code to improve readability and maintainability

The current code contains redundant logic and complex methods,
making it difficult to manage.

Refactor by splitting large functions, removing repeated code,
and enhancing variable names for clarity.

This improves code readability and reduces complexity for easier
future updates. No functional changes were made.
